src/client/visualizations/bar-chart/improved-bar-chart/bars/bars.tsx

```diff
@@ -48,21 +45,20 @@ const Bar: React.SFC<BarProps> = props => {
       : <StackedBar {...rest} model={model} />;
   }
   return showPrevious
-    ? <SingleTimeShiftBar {...rest} />
-    : <SingleBar {...rest} />;
+    ? <SingleTimeShiftBar {...rest} model={model} />
+    : <SingleBar {...rest} model={model} />;
```

Comment:
Would there be a difference if we did like this?
```suggestion
    : <SingleBar {...props} />;
```

Currently `{...rest} model={model}` makes me think there is some important operation done on `model` between taking it from props and passing it down to components 🤔

Reply:
In your example, TS can't infer narrower type of `model`. We must be explicit for TS. Of course we could stitch `rest` and `model` together again with spread, but surprisingly correct TSX syntax is `<StackedTimeShiftBar {... { ...rest, model }}  />`